racetrack_module.py

```diff
@@ -0,0 +1,588 @@
+                start = QgsPointXY(
+                    params['point_start'].x() + params['vec_normalized'].x() * dist,
+                    params['point_start'].y() + params['vec_normalized'].y() * dist,
+                )
+                end = QgsPointXY(
+                    params['point_end'].x() + params['vec_normalized'].x() * dist,
+                    params['point_end'].y() + params['vec_normalized'].y() * dist
+                )
+
+                if left_point:
+                    points.append(start)
+                    points.append(end)
+                else:
+                    points.append(end)
+                    points.append(start)
+
+                left_point = not left_point
+
+                if forward:
+                    if j + max_flyover <= number_of_lines:
+                        j = j + max_flyover
+                        inner_iteration = inner_iteration + 1
+                        if ((j - 1) / ((2.0 * max_flyover) - 1.0)) % 1.0 != 0.0:
+                            forward = not forward
+                        else:
+                            inner_iteration = 0
+                    else:
+                        reached_end = True
+                else:
+                    j = (j - max_flyover) + 1
+                    inner_iteration = inner_iteration + 1
+                    forward = not forward
+            else:
+                if inner_iteration == 0:
+                    remaining_tracks = number_of_lines - j
+                else:
+                    remaining_tracks = int((((2 * max_flyover) - inner_iteration) - 2) / 2)
+
+                for i in range(remaining_tracks):
+                    if forward:
+                        j = j + remaining_tracks
+                    else:
+                        j = j - remaining_tracks
+
+                    dist = 2 * params['coverage_range'] * params['overlap_factor'] * j - params['coverage_range']
+                    start = QgsPointXY(
+                        params['point_start'].x() + params['vec_normalized'].x() * dist,
+                        params['point_start'].y() + params['vec_normalized'].y() * dist,
+                    )
+                    end = QgsPointXY(
+                        params['point_end'].x() + params['vec_normalized'].x() * dist,
+                        params['point_end'].y() + params['vec_normalized'].y() * dist
+                    )
+
+                    if left_point:
+                        points.append(start)
+                        points.append(end)
+                    else:
+                        points.append(end)
+                        points.append(start)
+
+                    left_point = not left_point
+                    remaining_tracks = remaining_tracks - 1
+                    forward = not forward
+                    left_point = not left_point
+                break
+
+        return points
+
+    def _compute_fly_to_top_points(self, params: dict) -> List[QgsPointXY]:
```

Comment:
Method can be made static
```suggestion
    @staticmethod
    def _compute_fly_to_top_points(params: dict) -> List[QgsPointXY]:
```

Reply:
same as before